core/services/wifi/WifiManager.py

```diff
@@ -314,7 +323,23 @@ async def auto_reconnect(self, seconds_before_reconnecting: float) -> None:
                 self.connection_status = ConnectionStatus.STILL_CONNECTED
 
             if not networks_reenabled and seconds_disconnected >= seconds_before_reconnecting:
-                logger.debug("Watchdog activated. Trying to reconnect to available networks.")
+                logger.debug("Watchdog activated. Starting hotspot and trying to reconnect to available networks.")
                 await self.enable_saved_networks(self._ignored_reconnection_networks)
                 await self.wpa.send_command_reconnect()
+                self.enable_hotspot(True)
                 networks_reenabled = True
+
+    def enable_hotspot(self, enable: bool, ssid: Optional[str] = None, passphrase: Optional[str] = None) -> None:
+        if ssid is not None:
+            self._hotspot.set_ssid(ssid)
+        if passphrase is not None:
+            self._hotspot.set_passphrase(passphrase)
+        if enable:
```

Comment:
To make it more clear and easy to use, I would recommend to use a function just to disable/stop and remove the `enable` argument of the function